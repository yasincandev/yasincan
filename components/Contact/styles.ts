import styled from 'styled-components'

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 2em;
  list-style: none;
  font-family: 'Poppins', sans-serif;
  color: ${props => props.theme.helloColor};
  width: 100%;
  padding: 1em 0;

  @media (max-width: 768px) {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
  }
`

export const LinkContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  gap: 0.5em;

  cursor: pointer;
  transition: all 0.3s ease-in-out;
  width: 100%;
  text-align: center;
`

export const LinkText = styled.p`
  color: ${props => props.theme.greetingColor};
  font-size: 0.75rem;
`
