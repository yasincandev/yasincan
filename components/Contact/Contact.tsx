import Link from 'next/link'
import { FC } from 'react'
import { GithubIcon } from '../Icons/Github'
import { LinkedinIcon } from '../Icons/Linkedin'
import { ResumeIcon } from '../Icons/Resume'
import { Container, LinkContainer, LinkText } from './styles'
import { MediumIcon } from '../Icons/Medium'

const Contact: FC = () => {
  return (
    <Container>
      <Link
        href="https://linkedin.com/in/yasincandev"
        target="_blank"
        style={{ textDecoration: 'none' }}
      >
        <LinkContainer>
          <LinkedinIcon />
          <LinkText>Let&#39;s Connect! </LinkText>
        </LinkContainer>
      </Link>
      <Link
        href="https://github.com/yasincandev"
        target="_blank"
        style={{ textDecoration: 'none' }}
      >
        <LinkContainer>
          <GithubIcon />
          <LinkText>Review My Code</LinkText>
        </LinkContainer>
      </Link>
      <Link
        href="https://drive.google.com/file/d/1qAf-7f6HPLbHbzNaF6fK-W9xlC2GOjfz/view"
        target="_blank"
        style={{ textDecoration: 'none' }}
      >
        <LinkContainer>
          <ResumeIcon />
          <LinkText>Check Out My Resume</LinkText>
        </LinkContainer>
      </Link>
      <Link
        href="https://medium.com/@yasincandev"
        target="_blank"
        style={{ textDecoration: 'none' }}
      >
        <LinkContainer>
          <MediumIcon />
          <LinkText>My Blog</LinkText>
        </LinkContainer>
      </Link>
    </Container>
  )
}

export default Contact
