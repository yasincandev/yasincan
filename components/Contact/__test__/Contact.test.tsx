import { render } from '@testing-library/react'
import Contact from '../Contact'

it('should render Contact component', () => {
  const { container } = render(<Contact />)
  expect(container).toMatchSnapshot()
})
