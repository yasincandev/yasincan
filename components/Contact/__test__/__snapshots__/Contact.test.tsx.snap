// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Contact component 1`] = `
<div>
  <div
    class="sc-ed3a60c7-0 jLcnFB"
  >
    <div
      class="sc-ed3a60c7-1 MSzpr"
    >
      <a
        href="https://linkedin.com/in/yasincandev"
        style="text-decoration: none;"
        target="_blank"
      >
        <svg
          class="sc-d3a65e7b-0 bIojVO"
          height="100"
          viewBox="0 0 50 50"
          width="100"
          x="0"
          xmlns="http://www.w3.org/2000/svg"
          y="0"
        >
          <path
            d="M25 2C12.31 2 2 12.31 2 25s10.31 23 23 23 23-10.31 23-23S37.69 2 25 2zm0 2c11.61 0 21 9.39 21 21s-9.39 21-21 21S4 36.61 4 25 13.39 4 25 4zm-9 8c-1.645 0-3 1.355-3 3s1.355 3 3 3 3-1.355 3-3-1.355-3-3-3zm0 2c.564 0 1 .436 1 1 0 .564-.436 1-1 1-.564 0-1-.436-1-1 0-.564.436-1 1-1zm-2 5a1 1 0 00-1 1v15a1 1 0 001 1h4a1 1 0 001-1V20a1 1 0 00-1-1h-4zm8 0a1 1 0 00-1 1v15a1 1 0 001 1h4a1 1 0 001-1v-7.5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5V35a1 1 0 001 1h4a1 1 0 001-1v-8.5c0-4.13-3.37-7.5-7.5-7.5-1.286 0-2.44.416-3.5.99a1 1 0 00-1-.99h-4zm-7 2h2v13h-2V21zm8 0h2v.816a1 1 0 001.693.721A5.464 5.464 0 0130.5 21c3.05 0 5.5 2.45 5.5 5.5V34h-2v-6.5c0-2.47-2.03-4.5-4.5-4.5S25 25.03 25 27.5V34h-2V21z"
          />
        </svg>
        <p
          class="sc-ed3a60c7-2 gQzJFZ"
        >
          Let's Connect! 
        </p>
      </a>
    </div>
    <div
      class="sc-ed3a60c7-1 MSzpr"
    >
      <a
        href="https://github.com/yasincandev"
        style="text-decoration: none;"
        target="_blank"
      >
        <svg
          class="sc-f6b89e6-0 hKmXof"
          height="100"
          viewBox="0 0 64 64"
          width="100"
          x="0"
          xmlns="http://www.w3.org/2000/svg"
          y="0"
        >
          <path
            d="M32 10c12.15 0 22 9.85 22 22 0 9.768-6.369 18.045-15.179 20.916l.006-.021s-1.485-.696-1.453-1.938c.035-1.367 0-4.556 0-5.727 0-2.01-1.272-3.434-1.272-3.434s9.977.112 9.977-10.533c0-4.107-2.147-6.245-2.147-6.245s1.128-4.385-.39-6.245c-1.701-.184-4.749 1.626-6.05 2.472 0 0-2.062-.846-5.492-.846s-5.492.846-5.492.846c-1.301-.846-4.348-2.656-6.05-2.472-1.518 1.86-.39 6.245-.39 6.245s-2.147 2.137-2.147 6.245c0 10.645 9.977 10.533 9.977 10.533s-1.005 1.136-1.225 2.806c-.696.236-1.721.528-2.549.528-2.165 0-3.812-2.105-4.416-3.078-.595-.96-1.815-1.766-2.953-1.766-.749 0-1.115.375-1.115.803s1.05.727 1.743 1.521c1.461 1.674 1.435 5.438 6.641 5.438.565 0 1.719-.139 2.588-.256-.005 1.185-.007 2.436.012 3.167.031 1.242-1.453 1.938-1.453 1.938l.006.021C16.369 50.045 10 41.768 10 32c0-12.15 9.85-22 22-22z"
          />
        </svg>
        <p
          class="sc-ed3a60c7-2 gQzJFZ"
        >
          Review My Code
        </p>
      </a>
    </div>
    <div
      class="sc-ed3a60c7-1 MSzpr"
    >
      <a
        href="https://drive.google.com/file/d/1OxmrpVn40hrvqBhuluor-wMoVan1xEuj/view?usp=sharing"
        style="text-decoration: none;"
        target="_blank"
      >
        <svg
          class="sc-3b35e6d0-0 ylTJW"
          height="100"
          viewBox="0 0 50 50"
          width="100"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            d="M 7 2 L 7 15 L 9 17 L 9 4 L 42 4 L 42 46 L 9 46 L 9 38 L 7 40 L 7 48 L 44 48 L 44 2 Z M 17.40625 8 C 15.226563 8.039063 15.046875 9.730469 15.46875 11.359375 C 15.46875 11.359375 15.242188 11.589844 15.253906 11.910156 C 15.277344 12.523438 15.683594 12.71875 15.683594 12.71875 C 15.761719 13.289063 16.261719 13.796875 16.441406 13.890625 C 16.441406 14.253906 16.480469 14.53125 16.441406 14.929688 C 16.011719 16.089844 13.132813 15.761719 13 18 L 21.890625 18 C 21.753906 15.761719 18.875 16.089844 18.441406 14.929688 C 18.40625 14.53125 18.441406 14.253906 18.441406 13.890625 C 18.691406 13.773438 19.101563 13.296875 19.183594 12.71875 C 19.183594 12.71875 19.609375 12.445313 19.609375 11.996094 C 19.609375 11.679688 19.503906 11.445313 19.421875 11.359375 C 19.648438 10.664063 20.042969 8.550781 18.527344 8.550781 C 18.363281 8.265625 17.953125 8 17.40625 8 Z M 28 10 C 27.640625 9.996094 27.304688 10.183594 27.121094 10.496094 C 26.941406 10.808594 26.941406 11.191406 27.121094 11.503906 C 27.304688 11.816406 27.640625 12.003906 28 12 L 36 12 C 36.359375 12.003906 36.695313 11.816406 36.878906 11.503906 C 37.058594 11.191406 37.058594 10.808594 36.878906 10.496094 C 36.695313 10.183594 36.359375 9.996094 36 10 Z M 28 15 C 27.640625 14.996094 27.304688 15.183594 27.121094 15.496094 C 26.941406 15.808594 26.941406 16.191406 27.121094 16.503906 C 27.304688 16.816406 27.640625 17.003906 28 17 L 36 17 C 36.359375 17.003906 36.695313 16.816406 36.878906 16.503906 C 37.058594 16.191406 37.058594 15.808594 36.878906 15.496094 C 36.695313 15.183594 36.359375 14.996094 36 15 Z M 7 17.863281 L 7 24 L 0 24 L 0 31 L 7 31 L 7 37.136719 L 18.5625 27.5 Z M 9 22.136719 L 15.4375 27.5 L 9 32.863281 L 9 29 L 2 29 L 2 26 L 9 26 Z M 22 25 C 21.640625 24.996094 21.304688 25.183594 21.121094 25.496094 C 20.941406 25.808594 20.941406 26.191406 21.121094 26.503906 C 21.304688 26.816406 21.640625 27.003906 22 27 L 36 27 C 36.359375 27.003906 36.695313 26.816406 36.878906 26.503906 C 37.058594 26.191406 37.058594 25.808594 36.878906 25.496094 C 36.695313 25.183594 36.359375 24.996094 36 25 Z M 22 31 C 21.640625 30.996094 21.304688 31.183594 21.121094 31.496094 C 20.941406 31.808594 20.941406 32.191406 21.121094 32.503906 C 21.304688 32.816406 21.640625 33.003906 22 33 L 33 33 C 33.359375 33.003906 33.695313 32.816406 33.878906 32.503906 C 34.058594 32.191406 34.058594 31.808594 33.878906 31.496094 C 33.695313 31.183594 33.359375 30.996094 33 31 Z M 22 37 C 21.640625 36.996094 21.304688 37.183594 21.121094 37.496094 C 20.941406 37.808594 20.941406 38.191406 21.121094 38.503906 C 21.304688 38.816406 21.640625 39.003906 22 39 L 36 39 C 36.359375 39.003906 36.695313 38.816406 36.878906 38.503906 C 37.058594 38.191406 37.058594 37.808594 36.878906 37.496094 C 36.695313 37.183594 36.359375 36.996094 36 37 Z"
          />
        </svg>
        <p
          class="sc-ed3a60c7-2 gQzJFZ"
        >
          Check Out My Resume
        </p>
      </a>
    </div>
    <div
      class="sc-ed3a60c7-1 MSzpr"
    >
      <a
        href="https://medium.com/@yasincandev"
        style="text-decoration: none;"
        target="_blank"
      >
        <svg
          class="sc-1fe0643c-0 eqNTZW"
          height="100"
          viewBox="0 0 50 50"
          width="100"
          x="0"
          xmlns="http://www.w3.org/2000/svg"
          y="0"
        >
          <path
            d="M9.813 2A1 1 0 009 3v15H4v2H0v6h4v17c0 2.75 2.25 5 5 5h32c2.75 0 5-2.25 5-5V26h4v-6h-4v-2h-5V3c0-.55-.45-1-1-1H9.812zM11 4h28v14h-8.5a3.514 3.514 0 00-3.5 3.5c0 1.922 1.578 3.5 3.5 3.5H33c.566 0 1 .434 1 1 0 .566-.434 1-1 1H17c-.566 0-1-.434-1-1 0-.566.434-1 1-1h2.5c1.922 0 3.5-1.578 3.5-3.5S21.422 18 19.5 18H11zm4.813 4c-.551.05-.958.543-.907 1.094.051.55.543.957 1.094.906h18c.36.004.695-.184.879-.496a1.01 1.01 0 000-1.008c-.184-.312-.52-.5-.879-.496H15.812zm0 4c-.551.05-.958.543-.907 1.094.051.55.543.957 1.094.906h13c.36.004.695-.184.879-.496a1.01 1.01 0 000-1.008c-.184-.312-.52-.5-.879-.496H15.812zM6 20h13.5c.84 0 1.5.66 1.5 1.5s-.66 1.5-1.5 1.5H17c-1.645 0-3 1.355-3 3s1.355 3 3 3h16c1.645 0 3-1.355 3-3s-1.355-3-3-3h-2.5c-.84 0-1.5-.66-1.5-1.5s.66-1.5 1.5-1.5H44v23c0 1.668-1.332 3-3 3H9c-1.668 0-3-1.332-3-3zm-4 2h2v2H2zm44 0h2v2h-2zM11 33c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm-26 4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4 0c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm-21 4c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
          />
        </svg>
        <p
          class="sc-ed3a60c7-2 gQzJFZ"
        >
          My Blog
        </p>
      </a>
    </div>
  </div>
</div>
`;
